src/Build/Evaluation/Evaluator.cs

```diff
@@ -2076,6 +2102,17 @@ private void EvaluateImportElement(string directoryOfImportingFile, ProjectImpor
                 PerformDepthFirstPass(importedProjectRootElement);
             }
 
+            if (!importedAnything && hasCondition)
```

Comment:
Ah, yes, it would fail here: [https://github.com/Microsoft/msbuild/blob/4b50e21f36ccf87abf0ad0adc93f203d44a3d848/src/Build/Evaluation/Evaluator.cs#L2425-L2430](https://github.com/Microsoft/msbuild/blob/4b50e21f36ccf87abf0ad0adc93f203d44a3d848/src/Build/Evaluation/Evaluator.cs#L2425-L2430)

---
In reply to: [118308379](https://github.com/Microsoft/msbuild/pull/2109#discussion_r118308379) [](ancestors = 118308379)